### PR TITLE
MRG: Link _TimeViewer time slider with vertical time line

### PR DESCRIPTION
**Pull request by GuillaumeFavelier**
This PR adds a vertical line to indicate the current time on the `matplotlib` canvas.

I also did quite the refactor there as `_TimeViewer.__init__` was getting too long.

output

It's an item of #7162

### CODE REVIEW

File: mne/viz/_brain/_timeviewer.py [hunk 1]
```diff
@@ -753,6 +775,26 @@ def plot_time_course(self, hemi, vertex_id, color):
...
+        if not hasattr(self, "mpl_canvas"):
+            return
+        if isinstance(self.show_traces, bool) and self.show_traces:
+            # add time information
+            current_time = self.brain._current_time
+            xdata = [current_time, current_time]
+            if not hasattr(self, "time_line_height"):
+                self.time_line_height = np.max(self.brain._data["array"])
+            if not hasattr(self, "time_line"):
+                self.time_line = self.mpl_canvas.plot(
```
Comment (larsoner) [inline mne/viz/_brain/_timeviewer.py]: Why not just `axvline` here?

**Comment by GuillaumeFavelier** [comment]
> feel free to take a look and fix if need be.

Could not have done better.

I just added `to_time_index()` to `_Brain`

**Comment by larsoner** [comment]
@GuillaumeFavelier can we make that private and also get rid of `index_for_time`? Things like `index_for_time` seem like cruft left over from `PySurfer` that we don't want in the public interface of our new `Brain` class.

**Comment by GuillaumeFavelier** [comment]
@larsoner there are multiple places where I consider that `time_idx` is `int` (which was guaranteed by `index_for_time` with the default parameters). This might take more time than expected to refactor.

**Comment by larsoner** [comment]
Okay, might not be worth it then...

**Comment by GuillaumeFavelier** [comment]
I took care of it in the latest commit 363e81f, what do you think?

**Comment by larsoner** [comment]
Thanks @GuillaumeFavelier

**Pull request merged**